synapseclient/core/upload/multipart_upload_async.py

```diff
@@ -126,6 +127,22 @@
 DEFAULT_PART_SIZE = 8 * MB
 MAX_RETRIES = 7
 
+_thread_local = threading.local()
+
+
+@contextmanager
+def shared_progress_bar(progress_bar):
```

Comment:
Will this only be used in the multi part upload?

Reply:
For now - Yes. I may extract this with changing how downloads are working.